nucypher/characters/lawful.py

```diff
@@ -595,50 +597,77 @@ def get_treasure_map_from_known_ursulas(self, network_middleware, map_id):
 
         return treasure_map
 
-    def generate_work_orders(self, map_id, *capsules, num_ursulas=None, cache=False):
+    def work_orders_for_capsules(self,
+                                 *capsules,
+                                 alice_verifying_key: UmbralPublicKey,
+                                 map_id: str = None,
+                                 treasure_map: 'TreasureMap' = None,
+                                 num_ursulas: int = None,
+                                 ):
+
         from nucypher.policy.collections import WorkOrder  # Prevent circular import
 
-        try:
-            treasure_map_to_use = self.treasure_maps[map_id]
-        except KeyError:
-            raise KeyError(
-                "Bob doesn't have the TreasureMap {}; can't generate work orders.".format(map_id))
+        if treasure_map:
+            map_id = treasure_map.public_id()
+            treasure_map_to_use = treasure_map
+        else:
+            try:
+                treasure_map_to_use = self.treasure_maps[map_id]
+            except KeyError:
+                raise KeyError(
+                    "Bob doesn't have the TreasureMap {}; can't generate work orders.".format(map_id))
 
-        generated_work_orders = OrderedDict()
+        incomplete_work_orders = OrderedDict()
+        complete_work_orders = OrderedDict()
 
         if not treasure_map_to_use:
             raise ValueError(
                 "Bob doesn't have a TreasureMap to match any of these capsules: {}".format(
                     capsules))
 
         for node_id, arrangement_id in treasure_map_to_use:
```

Comment:
Can you add this commit (https://github.com/nucypher/nucypher/commit/a7d9d7cbbe963e384f993e18c6f1f8c4ec5772a8) to the EPIC? It seems this is a good scope for it.

Reply:
:heavy_check_mark: - cherry picked the commit on top.